tests/integration/integration-test.sh

```diff
@@ -1,238 +1,304 @@
+    TYPE=$(echo ${test_case} | jq -r ".type")
+    REF=$(echo ${test_case} | jq -r ".ref")
+    NAMESPACE=$(echo ${test_case} | jq -r ".namespace")
+    EXP_MSG=$(echo ${test_case} | jq -r ".expected_msg")
+    EXP_RES=$(echo ${test_case} | jq -r ".expected_result")
+    single_test "${ID}" "${TEST_CASE_TXT}" "${TYPE}" "${REF}" "${NAMESPACE}" "${EXP_MSG}" "${EXP_RES}"
+  done
+}
+
+### WORKLOAD TEST ####################################
+workload_test() { # WORKLOAD_KIND
+  export KIND=$1
+  export APIVERSION=$(kubectl api-resources | awk -v KIND=${KIND} '{ if($NF == ""KIND"") print $(NF-2);}')
+
+  # UNSIGNED
+  export TAG=unsigned
+  echo "::group::${KIND}_${APIVERSION}_${TAG}.yaml"
+  envsubst <tests/integration/workload-objects/${KIND}.yaml | cat
+  echo "::endgroup::"
+  single_test "w_${KIND}_${APIVERSION}_${TAG}" "Testing ${KIND} using ${APIVERSION} and ${TAG} image..." "workload" "${KIND}" "deafult" "Unable to find signed digest for image docker.io/securesystemsengineering/testimage:unsigned." "null"
+
+  # SIGNED
+  export TAG=signed
+  echo "::group::${KIND}_${APIVERSION}_${TAG}.yaml"
+  envsubst <tests/integration/workload-objects/${KIND}.yaml | cat
+  echo "::endgroup::"
+  single_test "w_${KIND}_${APIVERSION}_${TAG}" "Testing ${KIND} using ${APIVERSION} and ${TAG} image..." "workload" "${KIND}" "deafult" " created" "null"
+}
+
+### STRESS TEST ####################################
+stress_test() { #
+  NUMBER_OF_INSTANCES=100
+  echo -n 'Testing Connaisseur with complex requests...'
+  kubectl apply -f tests/integration/deployments/stresstest.yaml >output.log 2>&1 || true
+  if [[ ! ("$(cat output.log)" =~ 'deployment.apps/redis-with-many-instances created' && "$(cat output.log)" =~ 'pod/pod-with-many-containers created' && "$(cat output.log)" =~ 'pod/pod-with-many-containers-and-init-containers created' && "$(cat output.log)" =~ 'pod/pod-with-some-containers-and-init-containers created' && "$(cat output.log)" =~ 'pod/pod-with-coinciding-containers-and-init-containers created') ]]; then
+    echo -e ${FAILED}
+    echo "::group::Output"
+    cat output.log
+    echo "::endgroup::"
+    EXIT="1"
+  else
+    echo -e "${SUCCESS}"
+  fi
+
+  echo -n 'Cleaning up before second test...'
+  kubectl delete all -ltest=stresstest >/dev/null
+  echo -e "${SUCCESS}"
+
+  echo -n 'Testing Connaisseur with many requests...'
+  parallel --jobs 20 ./tests/integration/cause_load.sh {1} :::: <(seq ${NUMBER_OF_INSTANCES}) >output.log 2>&1 || true
+  NUMBER_CREATED=$(cat output.log | grep "deployment[.]apps/redis-[0-9]* created" | wc -l || echo "0")
+  if [[ ${NUMBER_CREATED} != "${NUMBER_OF_INSTANCES}" ]]; then
+    echo -e ${FAILED}
+    echo "::group::Output"
+    echo "Only ${NUMBER_CREATED}/${NUMBER_OF_INSTANCES} pods were created."
+    cat output.log
+    echo "::endgroup::"
+    EXIT="1"
+  else
+    echo -e "${SUCCESS}"
+  fi
+}
+
+### INSTALLING CONNAISSEUR ####################################
+install_make() {
```

Comment:
Sorry, probably my suggestion was bad: This reads like "please install make on my system". Can we do "action_via_tool" or "tool_action" instead? 😅